### PR TITLE
add 'make watch-frontend' and expand docs

### PR DESCRIPTION
**Pull request by silverwind**
Not perfect as it requires two terminals, but it's still builds much faster then running the full build. Also, the dev mode of webpack enables additional logging in the browser console, so this should be the go-to mode for frontend development.

### CODE REVIEW

**Review by everhopingandwaiting — APPROVED** [review]
need it

File: docs/content/doc/advanced/hacking-on-gitea.en-us.md [hunk 1]
```diff
@@ -128,10 +128,28 @@ make revive vet misspell-check
...
 
-Edit files in `web_src` and run the linter and build the files in `public`:
+For simple changes, edit files in `web_src`, run the build and start the server to test:
 
 ```bash
-make webpack
+make build && ./gitea
+```
+
+For more involved changes use the `watch-frontend` task to continuously rebuild files when their sources change. First, build and run the backend:
```
Comment (guillep2k) [inline docs/content/doc/advanced/hacking-on-gitea.en-us.md]: I'd add a suggestion about the `bindata` tag. I think it should not be present?

Comment (silverwind) [reply docs/content/doc/advanced/hacking-on-gitea.en-us.md]: Yes indeed, it must not be present. I initially thought not mentioning it is fine but it's better t be explicit about it.

**Comment by lafriks** [comment]
Imho built in option would be better

**Comment by silverwind** [comment]
@lafriks yes a single command would be preferred but doing that also opens up issue that the runtime depends on the existance `node_modules` which I don't really like because gitea is a single-binary distribution that should be able to run standalone.

**Comment by lafriks** [comment]
@silverwind not really, it could be special flag to gitea web command that you can run that would start npx webpack. Something like `./gitea web --dev`

**Comment by lafriks** [comment]
or we could even detect if gitea is started in development mode and `node_modules` directory exists in current working directory and only than start npx webpack in background

**Comment by silverwind** [comment]
Yeah that could be added later (and this task be removed again), but I still think it's worthwile to have now (at least it saves me from typing that long command every time 😉).

**Review by guillep2k — APPROVED** [review]

**Comment by guillep2k** [comment]
If I may, I'd rather not pollute the binary with dev options. It's not likely that anyone would _not_ have the source files when doing front-end development, unless of course they're only working on a UI theme; but even then, the `.less` files are source code too.

**Review by 6543 — APPROVED** [review]
add a Make target is way more better than a bigger gitea binary and more dependencys

**Comment by guillep2k** [comment]
Ping LG-TM

**Pull request merged**